src/Lean/Parser/Basic.lean

```diff
@@ -1424,7 +1424,7 @@ def longestMatchStep (left? : Option Syntax) (startSize startLhsPrec : Nat) (sta
   let prevLhsPrec   := s.lhsPrec
   let s             := s.restore prevSize startPos
   let s             := runLongestMatchParser left? startLhsPrec p c s
-  match compare previousScore (score s prio) with
+  match (let _ := @lexOrd; compare previousScore (score s prio)) with
```

Comment:
Is this the correct idiom then :) ?

Reply:
You can also do `local attribute [instance] lexOrd` before the function, or have a type alias `Lex A B` on which to unconditionally put the instance. But this way is fine too, or `(inst := lexOrd)` in `compare`. If this program were subject to proofs I would generally prefer `haveI` here to avoid creating the additional let binding in the generated term, but for this code that's not really a concern and the compiler will normalize the two variants the same anyway.

Reply:
I will merge as-is. I think the `haveI`/`letI` remark is important, we should discuss it in the next dev-meeting. We also get back to the `let` vs `let dep` discussion. It would be great to use the same style for `letI` (`let inl`?), and have it as part of the `let` family: `let`, `let rec`, `let dep`, `let mut`, ...